noxfile.py

```diff
@@ -352,314 +352,6 @@ def _run_with_coverage(session, *test_cmd):
 
 
-def _runtests(session, coverage, cmd_args):
-    # Create required artifacts directories
-    _create_ci_directories()
-    try:
-        if coverage is True:
-            _run_with_coverage(
-                session,
-                "coverage",
-                "run",
-                os.path.join("tests", "runtests.py"),
-                *cmd_args
-            )
-        else:
-            cmd_args = ["python", os.path.join("tests", "runtests.py")] + list(cmd_args)
-            env = None
-            if IS_DARWIN:
-                # Don't nuke our multiprocessing efforts objc!
-                # https://stackoverflow.com/questions/50168647/multiprocessing-causes-python-to-crash-and-gives-an-error-may-have-been-in-progr
-                env = {"OBJC_DISABLE_INITIALIZE_FORK_SAFETY": "YES"}
-            session.run(*cmd_args, env=env)
-    except CommandFailed:  # pylint: disable=try-except-raise
-        # Disabling re-running failed tests for the time being
-        raise
-
-        # pylint: disable=unreachable
-        names_file_path = os.path.join("artifacts", "failed-tests.txt")
-        session.log("Re-running failed tests if possible")
-        session.install(
-            "--progress-bar=off", "xunitparser==1.3.3", silent=PIP_INSTALL_SILENT
-        )
-        session.run(
-            "python",
-            os.path.join(
-                "tests", "support", "generate-names-file-from-failed-test-reports.py"
-            ),
-            names_file_path,
-        )
-        if not os.path.exists(names_file_path):
-            session.log(
-                "Failed tests file(%s) was not found. Not rerunning failed tests.",
-                names_file_path,
-            )
-            # raise the original exception
-            raise
-        with open(names_file_path) as rfh:
-            contents = rfh.read().strip()
-            if not contents:
-                session.log(
-                    "The failed tests file(%s) is empty. Not rerunning failed tests.",
-                    names_file_path,
-                )
-                # raise the original exception
-                raise
-            failed_tests_count = len(contents.splitlines())
-            if failed_tests_count > 500:
-                # 500 test failures?! Something else must have gone wrong, don't even bother
-                session.error(
-                    "Total failed tests({}) > 500. No point on re-running the failed tests".format(
-                        failed_tests_count
-                    )
-                )
-
-        for idx, flag in enumerate(cmd_args[:]):
-            if "--names-file=" in flag:
-                cmd_args.pop(idx)
-                break
-            elif flag == "--names-file":
-                cmd_args.pop(idx)  # pop --names-file
-                cmd_args.pop(idx)  # pop the actual names file
-                break
-        cmd_args.append("--names-file={}".format(names_file_path))
```

Comment:
perhaps `--names-file` in `tests/conftest.py` can be dropped as well?

Reply:
There are a few CLI flags, which were also migrated to PyTest for backwards compat reasons, that can be removed.
It should go in another stage/PR and needs a careful review since our CI process might rely on them.